### PR TITLE
Basic composition docs.

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: site/docs/composition.md [hunk 1]
```diff
@@ -5,4 +5,42 @@ title: Composing Layered & Multi-view Plots
 permalink: /docs/composition.html
 ---
 
-TODO Intro to composition and link to compositional operators
+With Vega-Lite, you cannot only create visualizations that consist of a single view. With Vega-Lite's *view composition algebra*, you can combine views into larger multiview displays. You can combine views through [faceting](facet.html), [layering](layer.html), [concatenation](concat.html), and [repeating](repeat.html).
```
Comment (arvind) [inline site/docs/composition.md]: "you cannot only" -> "you can not only". It's also missing a "but also" clause. Perhaps the following:

> With Vega-Lite, you can not only create single view visualizations, but also [facet], [layer], [concatenate], and [repeat] these views into larger, more complex multiview displays.

(I think we should consider dropping "algebra" language from the documentation. That seems fairly complex for docs).

File: site/docs/composition.md [hunk 2]
```diff
@@ -5,4 +5,42 @@ title: Composing Layered & Multi-view Plots
...
+
+## Faceting
+
+With the `facet` operator, you can partition the data and create a view for each subset to create a [trellis plot](https://en.wikipedia.org/wiki/Small_multiple). As a shortcut, Vega-Lite also has the [`column` and `row` encoding channels](encoding.html#facet) to quickly create a faceted view.
+
+Learn how to use it on the [facet page](facet.html).
+
+## Layering
+
+With the `layer` operator, you can stack multiple views on top of each other. This can be useful to add annotations to views. Vega-Lite automatically unions scale domains and combines axes and legends.
```
Comment (arvind) [inline site/docs/composition.md]: Perhaps avoid overloading the word "stack"? Alternatives could be place, arrange.

File: site/docs/composition.md [hunk 3]
```diff
@@ -5,4 +5,42 @@ title: Composing Layered & Multi-view Plots
...
+
+However, Vega-Lite can not enforce that a unioned domain is *semantically meaningful*. To prohibit layering of composite views with incongruent internal structures, the layer operator restricts its operands to be single or layered views.
```
Comment (arvind) [inline site/docs/composition.md]: Consider simplifying the language here? What does "semantically meaningful" or "incongruent internal structures" mean?

File: site/docs/concat.md [hunk 4]
```diff
@@ -5,5 +5,44 @@ title: Concatenating views
...
+
+## Horizontal Concatenation
+
+Put multiple views into a row by putting the specs for each view into an array and assign it to the `hconcat` property.
+
+{: .suppress-error}
+```json
+{
+  "hconcat": [
+    ...  // Specifications
```
Comment (arvind) [inline site/docs/concat.md]: Perhaps identify these as "Single-View Specifications"?

File: site/docs/composition.md (hunk 1 above)
Comment (domoritz) [inline site/docs/composition.md]: Done.

File: site/docs/composition.md (hunk 2 above)
Comment (domoritz) [inline site/docs/composition.md]: God point. Done.

File: site/docs/composition.md (hunk 3 above)
Comment (domoritz) [inline site/docs/composition.md]: Hehe, this was a copy from the Vega-Lite paper. I agree, that it's hard to swallow. I revised it.

File: site/docs/concat.md (hunk 4 above)
Comment (domoritz) [inline site/docs/concat.md]: We can concatenate arbitrary views. However, we can only layer single or layered views. I revised the docs concerning this point/

**Comment by domoritz** [comment]
@arvind, d you want to merge these?

**Pull request merged**